src/main/java/parser/DataSetFormatter.java

```diff
@@ -202,8 +202,24 @@ public final void scanCommaSeparatedData(String myStr) {
     }
 
     public static void main(String args[]) {
-        String func = "sum(@(x)x^3+2*x+1,3,@(x)cos(x+4),5,-6,7,-8,8)";//"sort(sin(2),cosh(4),-6,log(12,10),-12,2*sin(3),cos(4),34)";
+        //String func = "sum(@(x)x^3+2*x+1,3,@(x)cos(x+4),5,-6,7,-8,8)";//"sort(sin(2),cosh(4),-6,log(12,10),-12,2*sin(3),cos(4),34)";
```

Comment:
All the new  sum/sin expressions belongs to the https://github.com/gbenroscience/ParserNG/blob/master/src/test/java/parser/methods/ext/SumTest.java
If you build this PR on https://github.com/gbenroscience/ParserNG/pull/23 you will get even more tests

Reply:
Also all th e old code from main can be moved to to tests. But I can do this  later for you if you wish.